src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -336,32 +337,28 @@ protected unsafe void UnsafeWriteUnicodeChars(char* chars, int charCount)
 
         protected unsafe int UnsafeGetUnicodeChars(char* chars, int charCount, byte[] buffer, int offset)
         {
-            char* charsMax = chars + charCount;
-            while (chars < charsMax)
+            if (BitConverter.IsLittleEndian)
             {
-                char value = *chars++;
-                buffer[offset++] = (byte)value;
-                value >>= 8;
-                buffer[offset++] = (byte)value;
+                new ReadOnlySpan<char>(chars, charCount)
+                    .CopyTo(MemoryMarshal.Cast<byte, char>(buffer.AsSpan(offset)));
             }
+            else
+            {
+                char* charsMax = chars + charCount;
+                while (chars < charsMax)
+                {
+                    char value = *chars++;
+                    buffer[offset++] = (byte)value;
+                    buffer[offset++] = (byte)(value >> 8);
+                }
```

Comment:
I believe we can now vectorize this for big-endian platforms thanks to https://github.com/dotnet/runtime/issues/75901
```suggestion
                BinaryPrimitives.ReverseEndianness(new ReadOnlySpan<short>(chars, charCount),
                    MemoryMarshal.Cast<byte, short>(buffer.AsSpan(offset)));
```

Reply:
Good suggestion, I've updated the code